lib/ldclient-rb/ldclient.rb

```diff
@@ -48,12 +48,11 @@ def initialize(api_key, config = Config.default)
 
     def flush
       events = []
-      num_events = @queue.length
       begin
-        num_events.times do
+        while true
```

Comment:
While this is technically more correct, there's a slight disadvantage in performance since this relies on exception handling as "normal" behavior. Our integration harness calls flush a lot-- I don't think this will be a substantial performance hit, but we'll see.